static/app/views/ddm/utils/parseMetricWidgetsQueryParam.tsx

```diff
@@ -28,6 +29,14 @@ function parseStringParam(
   return typeof value === 'string' ? value : undefined;
 }
 
+function parseNumberParam(
+  widget: Record<string, unknown>,
+  key: string
+): number | undefined {
+  const value = widget[key];
+  return typeof value === 'number' ? value : undefined;
```

Comment:
nit: can this be NaN? 😅

Reply:
not sure tbh, can JSON.parse return NaN? Will check.

Reply:
Seems like the js JSON implementation does not allow for NaN. Will still add the check 👍